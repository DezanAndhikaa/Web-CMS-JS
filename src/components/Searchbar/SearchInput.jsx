

import React from 'react';
import { InputBase, Paper } from '@material-ui/core';
import { SearchIcon } from '../../assets/icons';
import './SearchInput.scss';

export default class SearchInput extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
    };
  }

  handleKeyUp = (event) => {
    this.setState({ value: event.target.value });
    if (event.keyCode === 13) {
      this.props.onSearch(this.state.value);
    } else {
      setTimeout(() => {
        this.props.onSearch(this.state.value);
      }, 1000);
    }
  }

  render() {
    const info = (this.props.displayMode === 'web' ? this.props.webInfo : this.props.generalInfo) || 'Search';
    return (
      <Paper className={this.props.className || 'search-input'} elevation={1}>
        <img src={SearchIcon} alt="" className="search-icon" />
        <InputBase
          className="search-text"
          placeholder={info}
          onKeyUp={this.handleKeyUp}
        />
      </Paper>
    );
  }
}
