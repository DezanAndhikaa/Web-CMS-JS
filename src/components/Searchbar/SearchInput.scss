@import '~themes/variables';
@import '~themes/functions';

.search-input{
    width: 62%;
    position: relative;
    background-color: #ffffff !important;
    border-radius: vw(10) !important;
    display: flex;
    align-items: center;
    height: vw(38.07);
    box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161) !important;
    margin: 0 0 0 3.4vw;
    .search-icon{
        margin-left: vw(20.97);
        margin-right: vw(18.17);
        color: #BABABA;
        height: vw(16) !important;
    }
    .search-icon:hover{
        background: none;
        color: #3C3B3B;
    }
    .search-text{
        flex: 1;
        font-family: 'Lato-Regular';
        font-style: normal;
        font-weight: normal;
        font-size: 14px;
        line-height: 17px;
    }
}

.search-input-rev{
    width: 100%;
    position: relative;
    background-color: #ffffff !important;
    border-radius: vw(10) !important;
    display: flex;
    align-items: center;
    height: 35px;
    box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.1) !important;
    margin: 0 0 0 1.5vw;
    .search-icon{
        margin-left: vw(20.97);
        margin-right: vw(18.17);
        color: #BABABA;
        height: vw(16) !important;
    }
    .search-icon:hover{
        background: none;
        color: #3C3B3B;
    }
    .search-text{
        flex: 1;
        font-family: 'Lato-Regular';
        font-style: normal;
        font-weight: normal;
        font-size: 14px;
        line-height: 17px;
    }
}

.search-input-status{
    width: 100%;
    position: relative;
    background-color: #ffffff !important;
    border-radius: vw(10) !important;
    display: flex;
    align-items: center;
    height: 35px;
    box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.1) !important;
    margin: 0 0 0 vw(22);
    .search-icon{
        margin-left: vw(20.97);
        margin-right: vw(18.17);
        color: #BABABA;
        height: vw(16) !important;
    }
    .search-icon:hover{
        background: none;
        color: #3C3B3B;
    }
    .search-text{
        flex: 1;
        font-family: 'Lato-Regular';
        font-style: normal;
        font-weight: normal;
        font-size: 14px;
        line-height: 17px;
    }
}

@media (min-width: 600px) and (max-width: 1025px) {
    .search-input{
        width: 100%;
        position: relative;
        background-color: #ffffff !important;
        border-radius: vw-tab(15) !important;
        display: flex;
        align-items: center;
        height: vw-tab(46);
        box-shadow: 0px vw-tab(3) vw-tab(6) rgba(0, 0, 0, 0.161) !important;
        .search-icon{
            margin-left: vw-tab(53);
            margin-right: vw-tab(20);
            color: #D3D3D3;
            height: vw-tab(20) !important;
        }
        .search-icon:hover{
            background: none;
            color: #3C3B3B;
        }
        .search-text{
            font-family: 'Lato-Regular';
            font-size: vw-tab(20);
            line-height: vw-tab(20);
            flex: 1;
        }
    }
}

@media (max-width: 599px) {
    .search-input{
        width: 100%;
        position: relative;
        background-color: #ffffff !important;
        border-radius: vw-mobile(15) !important;
        display: flex;
        align-items: center;
        height: vw-mobile(38);
        box-shadow: 0px vw-mobile(3) vw-mobile(6) rgba(0, 0, 0, 0.161) !important;
        .search-icon{
            margin-left: vw-mobile(22);
            margin-right: vw-mobile(20);
            color: #D3D3D3;
            height: vw-mobile(25) !important;
        }
        .search-icon:hover{
            background: none;
            color: #3C3B3B;
        }
        .search-text{
            font-family: 'Lato-Regular';
            font-size: vw-mobile(16);
            line-height: vw-mobile(20);
            flex: 1;
        }
    }
}