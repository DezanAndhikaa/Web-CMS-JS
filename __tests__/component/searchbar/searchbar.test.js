import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import SearchInput from "../../../src/components/Searchbar/SearchInput";

Enzyme.configure({adapter:new Adapter()});
it("render searchbar correctly", () => {
  const wrapper = shallow (
    <SearchInput />
  )
  expect(wrapper).toMatchSnapshot();
})
