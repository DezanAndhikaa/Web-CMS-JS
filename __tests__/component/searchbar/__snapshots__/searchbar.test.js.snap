exports[`render searchbar correctly 1`] = `ShallowWrapper {}`;